src/libs/actions/TaxRate.ts

```diff
@@ -0,0 +1,130 @@
+import Onyx from 'react-native-onyx';
+import * as API from '@libs/API';
+import type {CreatePolicyTaxParams} from '@libs/API/parameters';
+import {WRITE_COMMANDS} from '@libs/API/types';
+import CONST from '@src/CONST';
+import * as ErrorUtils from '@src/libs/ErrorUtils';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {TaxRate, TaxRates} from '@src/types/onyx';
+import type {PendingAction} from '@src/types/onyx/OnyxCommon';
+import type {OnyxData} from '@src/types/onyx/Request';
+
+/**
+ * Get tax value with percentage
+ */
+function getTaxValueWithPercentage(value: string): string {
+    return `${value}%`;
+}
+
+function covertTaxNameToID(name: string) {
+    return `id_${name.toUpperCase().replaceAll(' ', '_')}`;
+}
+
+/**
+ * Get new tax ID
+ */
+function getNextTaxID(name: string, taxRates?: TaxRates): string {
+    const newID = covertTaxNameToID(name);
+    if (!taxRates?.[newID]) {
+        return newID;
+    }
+
+    // If the tax ID already exists, we need to find a unique ID
+    let nextID = 1;
+    while (nextID < 100) {
+        const newNameCandidate = covertTaxNameToID(`${name}_${nextID}`);
+        if (!taxRates?.[newNameCandidate]) {
+            return newNameCandidate;
+        }
+        nextID++;
+    }
+    throw new Error('Could not find a unique tax ID');
+}
+
+function createWorkspaceTax(policyID: string, taxRate: TaxRate) {
+    const onyxData: OnyxData = {
+        optimisticData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: {
+                            [taxRate.code]: {
+                                ...taxRate,
+                                pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+                                errors: null,
+                            },
+                        },
+                    },
+                },
+            },
+        ],
+        successData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: {
+                            [taxRate.code]: {
+                                pendingAction: null,
+                                errors: null,
+                            },
+                        },
+                    },
+                },
+            },
+        ],
+        failureData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: {
+                            [taxRate.code]: {
+                                pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
```

Comment:
@luacmartins @kosmydel  I think we need to set `pendingAction: null` in the failureData

Reply:
@DylanDylann in this way, I wouldn't know in the `clearTaxRateError` function whether the user wanted to update or create a new tax rate. If the user fails to create a new tax rate, we should clear it from the list. Otherwise, we just need to dismiss the error.

Reply:
+1 this has to be removed, user will remove the error using the red brick road and clicking X